ostd/src/mm/tlb.rs

```diff
@@ -207,21 +196,33 @@ cpu_local! {
     static FLUSH_OPS: SpinLock<OpsStack, LocalIrqDisabled> = SpinLock::new(OpsStack::new());
     /// Whether this CPU finishes the last remote flush request.
     static ACK_REMOTE_FLUSH: AtomicBool = AtomicBool::new(true);
+
+    // Local buffers.
+    static LOCAL_FLUSH_OPS: RefCell<OpsStack> = RefCell::new(OpsStack::new());
 }
 
 fn do_remote_flush() {
-    let current_cpu = crate::cpu::current_cpu_racy(); // Safe because we are in IRQs.
+    // Safe because we are in IRQs/have disabled preempts.
+    let current_cpu = crate::cpu::current_cpu_racy();
+
+    let mut new_op_queue = OpsStack::new();
+    {
+        let mut op_queue = FLUSH_OPS.get_on_cpu(current_cpu).lock();
 
-    let mut op_queue = FLUSH_OPS.get_on_cpu(current_cpu).lock();
-    op_queue.flush_all();
+        core::mem::swap(&mut *op_queue, &mut new_op_queue);
 
-    ACK_REMOTE_FLUSH
-        .get_on_cpu(current_cpu)
-        .store(true, Ordering::Release);
+        // ACK before dropping the lock so that we won't miss flush requests.
+        ACK_REMOTE_FLUSH
+            .get_on_cpu(current_cpu)
+            .store(true, Ordering::Release);
```

Comment:
This `Release` _does_ pair with an `Acquire` in the `sync_tlb_flush` method.

But I wonder if there is any important information to synchronize here? If not, I think converting both to `Relaxed` might be more correct?